File: nomad-samples/portal/py/media_api/media/media_search.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def media_search(AUTH_TOKEN: str, SEARCH_QUERY: str, IDS: list, SORT_FIELDS_NAME: str, 
                 SORT_FIELDS_ORDER: str) -> dict:

    # Check for valid parameters
    if (not AUTH_TOKEN):
        raise Exception("Authentication Token: The authentication token is invalid")

    API_URL = PORTAL_API_URL + "/portal/search"
        
    # Create header for the request
    HEADERS = {
  	    "Authorization": "Bearer " + AUTH_TOKEN,
        "Content-Type": "application/json"
    }

    # Build the payload BODY
    BODY = {  
        "searchQuery": SEARCH_QUERY,
        "ids": IDS, 
        "sortFields": [  
            {  
                "fieldName": SORT_FIELDS_NAME,
                "sortType": SORT_FIELDS_ORDER  
            }  
        ]  
    }

    try:
        # Send the request
        RESPONSE = requests.post(API_URL, headers= HEADERS, data= json.dumps(BODY))

        if not RESPONSE.ok:
            raise Exception()

        return json.loads(RESPONSE.text)

    except:
        api_exception_handler(RESPONSE, "Media search failed")

File: nomad-samples/portal/js/media-api/media/media-search.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function mediaSearch(AUTH_TOKEN, SEARCH_QUERY, IDS, FIELD_NAME, SORT_TYPE) 
{
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
  
    const BODY = {  
        searchQuery: SEARCH_QUERY,
        ids: IDS,
        sortFields: [  
            {  
                fieldName: FIELD_NAME,
                sortType: SORT_TYPE  
            }  
        ]  
    };

    // Post
    const RESPONSE = await fetch(`${prjConstants.PORTAL_API_URL}/media/search`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the JSON from the response
        const INFO = await RESPONSE.json();

        return INFO;
    }
		
  	// There was an error
    await apiExceptionHandler(RESPONSE, "Media search failed");
}